src/pyo3_extensions/mod.rs

```diff
@@ -115,13 +115,18 @@ impl serde::Serialize for TdPyAny {
     where
         S: serde::Serializer,
     {
-        let bytes: Result<Vec<u8>, PyErr> = Python::with_gil(|py| {
+        Python::with_gil(|py| {
             let x = self.as_ref(py);
-            let pickle = py.import("dill")?;
-            let bytes = pickle.call_method1("dumps", (x,))?.extract()?;
-            Ok(bytes)
-        });
-        serializer.serialize_bytes(bytes.map_err(S::Error::custom)?.as_slice())
+            let pickle = py.import("dill").map_err(S::Error::custom)?;
```

Comment:
I think you could get away with only one `map_err` if you use a closure and immediately call it (one day we can use [try blocks](https://doc.rust-lang.org/beta/unstable-book/language-features/try-blocks.html)) in order to "capture" the `PyResult` error type first. It's a little annoying that you'll probably need type annotations, though. Not sure if this is more clear or not.

```rust
Python::with_gil(|py| {
    let pybytes: PyResult<PyBytes> = || {
        let pickle = ...?;
        pickle...
            .downcast::<PyBytes>()?
    }();
    serializer.serialize_bytes(bytes.map_err(S::Error::custom)?.as_bytes())
})
```